### PR TITLE
Add capacity and corresponding business rules

### PR DESCRIPTION
**Pull request by kft**

### CODE REVIEW

File: source/Energinet.DataHub.MeteringPoints.Tests/Domain/MeteringPoints/MarketMeteringPoints/MarketMeteringPointTests.cs [hunk 1]
```diff
@@ -70,25 +70,29 @@ public void Should_return_error_when_geo_info_reference_is_undefined()
...
-                ConnectionType.Installation,
-                AssetType.Boiler,
-                ScheduledMeterReadingDate.Create("0101"));
-
+            var details = CreateDetails()
+                with
+                {
+                    Address = address,
+                };
+            // var details = new MeteringPointDetails(
```
Comment (Korgath) [inline source/Energinet.DataHub.MeteringPoints.Tests/Domain/MeteringPoints/MarketMeteringPoints/MarketMeteringPointTests.cs:78]: Outcommented code

File: source/Energinet.DataHub.MeteringPoints.Tests/Domain/MeteringPoints/CapacityTests.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace Energinet.DataHub.MeteringPoints.Tests.Domain.MeteringPoints
+{
+    public class CapacityTests : TestBase
+    {
+        [Theory]
+        [InlineData("127.7777777", true)]
+        [InlineData("1277777777", true)]
+        [InlineData("ACBEE", true)]
+        [InlineData("127777777", false)]
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.Tests/Domain/MeteringPoints/CapacityTests.cs:28]: Will produce error once length comment above is fixed.

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Capacity.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+using Energinet.DataHub.MeteringPoints.Domain.MeteringPoints.Rules;
+using Energinet.DataHub.MeteringPoints.Domain.SeedWork;
+
+namespace Energinet.DataHub.MeteringPoints.Domain.MeteringPoints
+{
+    public class Capacity : ValueObject
+    {
+        private Capacity(float kwh)
+        {
+            Kwh = kwh;
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Capacity.cs:27]: guide says kW:
image

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Rules/CapacityFormatRule.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            }
+        }
+
+        public bool IsBroken { get; }
+
+        public ValidationError ValidationError { get; }
+
+        private static int GetLength(string capacityInKwh)
+        {
+            return !capacityInKwh.Contains('.', StringComparison.OrdinalIgnoreCase) ? capacityInKwh.Length : capacityInKwh.Remove(capacityInKwh.IndexOf('.', StringComparison.OrdinalIgnoreCase), 1).Length;
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Rules/CapacityFormatRule.cs:49]: Pretty sure the decimal point should count as a character. See pic above.

**Review by erduna — CHANGES_REQUESTED** [review]

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Rules/CapacityFormatRule.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+using System;
+using System.Globalization;
+using Energinet.DataHub.MeteringPoints.Domain.SeedWork;
+
+namespace Energinet.DataHub.MeteringPoints.Domain.MeteringPoints.Rules
+{
+    public class CapacityFormatRule : IBusinessRule
+    {
+        private const int MaxLength = 9;
```
[the remark this answers is not in the document]
Comment (erduna) [reply source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Rules/CapacityFormatRule.cs:23]: aparantly changed to 9 in unreleased documentation

File: source/Energinet.DataHub.MeteringPoints.Tests/Domain/MeteringPoints/MarketMeteringPoints/MarketMeteringPointTests.cs (hunk 1 above)
Comment (kft) [reply source/Energinet.DataHub.MeteringPoints.Tests/Domain/MeteringPoints/MarketMeteringPoints/MarketMeteringPointTests.cs:78]: Removed in next PR :-)

**Review by Korgath — DISMISSED** [review]

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Rules/CapacityFormatRule.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+        public CapacityFormatRule(string capacityInKwh)
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Rules/CapacityFormatRule.cs:25]: here 
```suggestion
        public CapacityFormatRule(string capacityInKw)
```

**Review by Korgath — APPROVED** [review]

**Pull request merged**